add.php

```diff
@@ -0,0 +1,144 @@
+<?php
+    require_once('config.php');
+    require_once('helpers.php');
+    require_once('functions.php');
+    require_once('validation.php');
+
+    $tabTypeId = filter_input(INPUT_GET, 'type_id', FILTER_SANITIZE_NUMBER_INT) ?? 1;
+
+    $connection = getConnection(DBHOST, DBUSER, DBPASSWORD, DBNAME);
+    $contentTypes = getContentTypes($connection);
+
+    $errors = [];
+    $rules = [
+        'heading' => function () {
+            return validateTitle('heading');
+        },
+        'text' => function () {
+            return validateText('text');
+        },
+        'author' => function () {
+            return validateAuthor('author');
+        },
+        'url' => function () {
+            return validateUrl('url');
+        },
+    ];
+
+    if (count($_POST) > 0) {
+        $title = $_POST[ 'heading' ] ?? '';
+        $text = $_POST[ 'text' ] ?? '';
+        $author = $_POST[ 'author' ] ?? '';
+        $video = $_POST[ 'typeName' ] === 'video' ? $_POST[ 'url' ] : '';
+        $link = $_POST[ 'typeName' ] === 'link' ? $_POST[ 'url' ] : '';
+        $image = '';
+        $tags = [];
+        $userId = '2';
+        $typeId = $_POST[ 'typeId' ];
+
+        foreach ($_POST as $key => $value) {
+            if (isset($rules[ $key ])) {
+                $rule = $rules[ $key ];
+                $errors[ $key ] = $rule();
+            }
+
+            if ($key === 'tags' && !empty($value)) {
+                $tagsArray = explode(' ', $value);
+                $tags = array_diff($tagsArray, array(''));
```

Comment:
```suggestion
                $tags = array_diff($tagsArray, []);
```
Начиная с php 5.6 можно использовать сокращённый синтаксис массивов